// Copyright (C) 2023 The Android Open Source Project
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package controlenvproxyserver;

import "google/protobuf/empty.proto";

service ControlEnvProxyService {
  rpc CallUnaryMethod (CallUnaryMethodRequest) returns (CallUnaryMethodReply) {}
  rpc ListServices (google.protobuf.Empty) returns (ListServicesReply) {}
  rpc ListMethods (ListMethodsRequest) returns (ListMethodsReply) {}
  rpc ListReqResType (ListReqResTypeRequest) returns (ListReqResTypeReply) {}
  rpc TypeInformation (TypeInformationRequest) returns (TypeInformationReply) {}
}

message CallUnaryMethodRequest {
  string service_name = 1;
  string method_name = 2;
  string json_formatted_proto = 3;
}

message CallUnaryMethodReply {
  string json_formatted_proto = 1;
}

message ListServicesReply {
  repeated string services = 1;
}

message ListMethodsRequest {
  string service_name = 1;
}

message ListMethodsReply {
  repeated string methods = 1;
}

message ListReqResTypeRequest {
  string service_name = 1;
  string method_name = 2;
}

message ListReqResTypeReply {
  string request_type_name = 1;
  string response_type_name = 2;
}

message TypeInformationRequest {
  string service_name = 1;
  string type_name = 2;
}

message TypeInformationReply {
  string text_formatted_type_info = 1;
}
